doc/nrf/ug_thread_vendor_hooks.rst

```diff
@@ -0,0 +1,229 @@
+You can use the available commands and properties, but also define your set of instructions to provide new features by using the vendor hooks.
+
+.. note::
+    Both devices used for communication must support user-defined commands and properties to achieve the expected result.
+    See the following sections for information about how to add your features on the host and the NCP/RCP devices.
+
+
+ID range of general vendor properties
+*************************************
+
+Spinel protocol associates specific properties with its numeric identifiers.
+The ID range reserved for the vendor properties goes from ``0x3c00`` to ``0x4000``.
+
+You are not allowed to use IDs from outside this range for your properties, as this may negatively affect the Spinel protocol operation and the vendor hooks mechanism will not work.
+
+
+Host device configuration
+*************************
+
+In most cases, the host device uses tools that support the Spinel protocol (like `wpantund`_ or `pyspinel`_) to communicate with the NCP/RCP device.
+You can, however, use another tool as long as it supports the Spinel protocol.
+
+Adding new property in pyspinel
+===============================
+
+pyspinel is the recommended tool for adding new properties, given its ease-of-use.
+
+The configuration of new properties in pyspinel is centered around the following files:
+
+* :file:`const.py` - This file contains ``VENDOR_SPINEL`` class, which stores definitions of the identifier values of vendor properties.
+  Use this file to add new properties.
+* :file:`codec.py` - This file contains definitions of handler methods called on the property detection.
+  Use this file to add definitions of handlers for your properties.
+* :file:`vendor.py` - This file contains methods to capture property occurrences from the console and call proper handlers.
+
+These files are located in the vendor directory, which can be found in the main project directory.
+
+To add a new property in pyspinel:
+
+1. Clone the `pyspinel`_ respository and install the tool.
+#. Navigate to the vendor directory with the :file:`codec.py`, :file:`const.py` and :file:`vendor.py` files.
+   This directory is located in the main project directory.
+#. Open the :file:`const.py` file for editing.
+#. Add new properties in the :file:`const.py` file.
+   For example, add the following properties:
+
+   * ``PROP_VENDOR_NAME`` that is used to get the vendor name from the device.
+   * ``PROP_VENDOR_AUTO_ACK_ENABLED`` that will be used to get or set the auto ACK mode state.
+
+   The class should contain code similar to the following:
+
+   .. code-block:: python
+
+      class VENDOR_SPINEL(object):
+        """
+        Class to extend SPINEL constant variables for example:
+            PROP_VENDOR__BEGIN = 0x3C00
+
+            PROP_VENDOR_HOOK = PROP_VENDOR__BEGIN + 0
+            PROP_VENDOR__END = 0x4000
+        """
+        PROP_VENDOR_NAME = 0x3c11
```

Comment:
Isn't there a predefined value mentioned like in the comment eg. PROP_VENDOR_NAME = PROP_VENDOR__BEGIN +0x11. It would be easier to read this example

Reply:
It is not, but it can be defined to make it look clearly. Good point. Done.